src/jit/gentree.cpp

```diff
@@ -6746,42 +7009,36 @@ void GenTreeIntCon::FixupInitBlkValue(var_types asgType)
 
 // Initializes a BlkOp GenTree
 // Preconditions:
-//     - Result is a GenTreeBlkOp that is newly constructed by gtNewCpObjNode or gtNewBlkOpNode
+//     - Result is an assignment that is newly constructed
 //
 // Parameters:
-//     - result is a GenTreeBlkOp node that is the node to be initialized.
-//     - oper must be either GT_INITBLK or GT_COPYBLK
+//     - result is an assignment node that is to be initialized.
 //     - dst is the target (destination) we want to either initialize or copy to
```

Comment:
No, the lhs may have been modified to a different type. However, I can add a comment and assert that if it is TYP_STRUCT, the lhs must be a block node or lclVar.

---
In reply to: [77389617](https://github.com/dotnet/coreclr/pull/7027#discussion_r77389617) [](ancestors = 77389617)